### PR TITLE
Add documentation comments to `bevy_winit`

### PR DESCRIPTION
**Pull request by beeryt**
# Objective

- [x] Add documentation comments to `bevy_winit`
- [x] Add `#![warn(missing_docs)]` to `bevy_winit`.

Relates to #3492

### CODE REVIEW

File: crates/bevy_winit/src/winit_windows.rs [hunk 1]
```diff
@@ -20,18 +21,24 @@ use crate::{
     converters::convert_window_level,
 };
 
+/// A reource which maps window entities to [`winit`] library windows.
```
Comment (mockersf) [inline crates/bevy_winit/src/winit_windows.rs:24]:
```suggestion
/// A resource which maps window entities to [`winit`] library windows.
```

**Comment by mockersf** [comment]
it seems you missed adding `#![warn(missing_docs)]`?

**Comment by beeryt** [comment]
> it seems you missed adding `#![warn(missing_docs)]`?

It appears that I did... and I'm also missing comments for the crate and `accessibility` module. Not sure at the moment what these should be. Tracking it in [Add missing doc comments](beeryt/bevy#1).

**Review by cart — APPROVED** [review]

File: crates/bevy_winit/src/winit_windows.rs [hunk 2]
```diff
@@ -20,18 +21,24 @@ use crate::{
     converters::convert_window_level,
 };
 
+/// A resource which maps window entities to [`winit`] library windows.
 #[derive(Debug, Default)]
 pub struct WinitWindows {
```
Comment (Vrixyz) [inline crates/bevy_winit/src/winit_windows.rs:26]: Sorry to come bakc after this PR is merged; but is this documentation accurate ? as the struct doesn't derive `Resource`, it's not a resource 🤔 ?

Comment (Vrixyz) [reply crates/bevy_winit/src/winit_windows.rs:26]: oh it's a `NonSendMut` :+1:

**Pull request merged**